occurrences/models.py

```diff
@@ -495,6 +497,23 @@ def with_enrolments_count(self, use_name_only=False):
             )
         return self.annotate(enrolments_count=Count("enrolments", distinct=True))
 
+    def user_can_view(self, user: User):
+        if not user.person:
+            return self.none
+        organisation_ids = [
+            entry[0] for entry in user.person.organisations.values_list("id")
+        ]
+        return self.with_organisation_ids().filter(
+            organisation_ids__contains=organisation_ids
+        )
```

Comment:
I tried to see what query this generates but was unable as running ```python
str(self.with_organisation_ids().filter(organisation_ids__contains=organisation_ids).query)
```
here gives
```python
*** django.core.exceptions.FieldError: Cannot resolve expression type, unknown output_field
```

I was trying to check whether this filter works so that the list on the left has to contain all of the values from the list on the right to pass through or not. I would think so, so then a question arises: Is this necessary or is this a too strict check? Should we let it through if at least a single value from the list on the right is found in the list on the left? How do you see this?

Reply:
Perhaps PostgreSQL specific [overlap](https://docs.djangoproject.com/en/3.2/ref/contrib/postgres/fields/#overlap) functionality could be used here:

```python
        return self.with_organisation_ids().filter(
            organisation_ids__overlap=organisation_ids
        )
```
?

Then this would pass AFAIK if at least a single value from the list on the right is found in the list on the left.

Reply:
This boils down to is the current filtering the correct one or does making it less strict actually let through data that we shouldn't let the user see?

Reply:
I mean, I'm OK to change it, I just don't understand the difference here. Is the problem that the aggregated value is not  an arrayField?

Reply:
Yea, so, the overlap really is the better option and it's now changed. Overlap means that "any of the items occurs" and contains meant that "all needs to occur".

Reply:
Nice catch @karisal-anders !